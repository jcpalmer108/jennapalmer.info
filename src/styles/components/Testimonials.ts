import styled from "styled-components"
import { off_black, redstone, silver_polish } from "../variables"

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
`

export const Content = styled.div`

`

export const Quote = styled.p`
  margin: 30px 0px;
  font-size: 15px;
  font-family: Gilroy-Regular;
  height: 130px;

`

export const Person = styled.div`
  font-size: 16;
  font-family: Gilroy-Medium;
  margin-bottom: 10px;

`

export const Title = styled.div`
  font-size: 12px;
  font-family: Gilroy-Regular;
  margin-bottom: 20px;

`

export const Dots = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`

type DotProps = {
  selected: boolean
}

export const Dot = styled.div<DotProps>`
  height: 7px;
  width: 7px;
  margin: 0px 7.5px;
  background: ${props => props.selected ? off_black : silver_polish};
  border-radius: 50%;
`


export const Arrows = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  column-gap: 20px;
  margin-top: 30px;

`

type ArrowProps = {
  desktop?: boolean | undefined
}

export const ArrowLeft = styled.img<ArrowProps>`
  display: ${props => props.desktop ? 'none' : 'block'}; 
  width: 19px;
`

export const ArrowRight = styled.img<ArrowProps>`
  display: ${props => props.desktop ? 'none' : 'block'};
  width: 19px;
`

export const Circle = styled.div`
  background: ${redstone};
  opacity: 10%;
  height: 40px;
  width: 40px;
  border-radius: 50%;
  z-index: -1;
  position: absolute;
`
export const Option = styled.div`
  height: 40px;
  width: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
`