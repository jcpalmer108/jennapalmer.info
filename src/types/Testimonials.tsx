import { Sections } from "./content"

export type TestimonialProps = {
  content: Sections | undefined
}