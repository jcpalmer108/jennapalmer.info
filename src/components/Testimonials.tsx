import { TestimonialProps } from "../types/Testimonials"
import { 
  Wrapper, 
  Quote, 
  Person, 
  Title, 
  Dots, 
  ArrowLeft, 
  ArrowRight, 
  Arrows, 
  Circle,
  Option,
  Dot,
  Content
} from "../styles/components/Testimonials"
import Section from "./Section"
import arrowLeft from "../assets/images/arrow-left-pink.svg"
import arrowRight from "../assets/images/arrow-right-pink.svg"
import { useEffect, useState } from "react"
// import { animated, useSpring } from "react-spring";

export default function Testimonials({ content }: TestimonialProps) {
  let [ selectedIndex, setSelectedIndex ] = useState(0)
  const quotes = content?.subSection?.quotes

  function updateIndex (moveForward: boolean) {
    if(moveForward) {
      return quotes && selectedIndex >= (quotes.length - 1) ? 
        setSelectedIndex(0) : 
        setSelectedIndex(selectedIndex + 1);
    } else {
      return quotes && selectedIndex <= 0 ? 
      setSelectedIndex(quotes.length - 1) : 
      setSelectedIndex(selectedIndex - 1);  
    }
  }
  // TODO: every 10 seconds, fade (quote, person, title) out, increase the index, then fade back in
  // TODO: tablet view
  // TODO: desktop view
  
  return (
    <Section>
      <Wrapper id="testimonials" data-testid="Testimonials">
        <ArrowLeft 
          onClick={() => { updateIndex(false) }} 
          src={arrowLeft} 
          alt="arrow left"
          data-testid="arrow left"
          desktop 
        />
        <Content>
          <h2>Testimonials</h2>
          <Quote data-testid="Quote">{quotes && quotes[selectedIndex].quote}</Quote>
          <Person data-testid="Person">{quotes && quotes[selectedIndex].person}</Person>
          <Title data-testid="Title">{quotes && quotes[selectedIndex].title}</Title>
        </Content>
          <Dots data-testid="Dots">
            {quotes && quotes.map((item, index) => {
              return (<Dot selected={selectedIndex === index}/>)
            })}
          </Dots>
        <ArrowRight 
          onClick={() => { updateIndex(true) }} 
          src={arrowRight} 
          alt="arrow right" 
          data-testid="arrow right"
          desktop 
        />
        <Arrows>
          <Option onClick={() => { updateIndex(false) }} >
            <Circle />
            <ArrowLeft 
              src={arrowLeft} 
              alt="arrow left" 
              data-testid="arrow left"
            />
          </Option>
          <Option onClick={() => { updateIndex(true) }}>
            <Circle />
            <ArrowRight data-testid="arrow right" src={arrowRight} alt="arrow right"/>
          </Option>
        </Arrows>
      </Wrapper>
    </Section>
  )
  
}