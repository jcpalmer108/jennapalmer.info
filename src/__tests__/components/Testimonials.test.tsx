import { render, screen } from '@testing-library/react'
import Testimonials from '../../components/Testimonials.tsx'

const params = {
  key: "testimonials",
  label: "Testimonials",
  title: "Test title",
  description: [
    "test description"
  ],
  inMenu: true,
  subSection: {
    quotes: [
      {
        quote: "this is a test quote",
        person: "Jane Doe",
        title: "Manager at Slalom Conuslting LLC"      
      },
      {
        quote: "this is a test quote",
        person: "John Doe",
        title: "Analyst at Slalom Conuslting LLC"      
      },
    ]
  }
}

describe('Testimonials', () => {
  test('renders if only required params are passed in', () => {
    <Testimonials content={params} />

    expect(screen.getByTestId('Testimonials')).toMatchSnapshot()
    expect(screen.getByTestId('Testimonials')).toMatchSnapshot()
    expect(screen.getByTestId('Marquee').childNodes).toHaveLength(params.subSection.quotes.length)

    params.subSection.quotes.forEach((item, index) => {
      expect(screen.getByTestId(`Quote ${index + 1}`)).toBeTruthy()
      expect(screen.getByTestId(`Quote ${index + 1}`).childNodes[0]).toHaveTextContent(item.quote)
      expect(screen.getByTestId(`Quote ${index + 1}`).childNodes[1]).toHaveTextContent(item.person)
      expect(screen.getByTestId(`Quote ${index + 1}`).childNodes[0]).toHaveTextContent(item.title)
    })

    expect(screen.getByRole('image', { name: 'arrow left' })).toBeTruthy()
    expect(screen.getByRole('image', { name: 'arrow right' })).toBeTruthy()
  })
})