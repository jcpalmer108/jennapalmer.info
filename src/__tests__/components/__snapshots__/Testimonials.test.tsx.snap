// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials renders if only required params are passed in 1`] = `
<div
  class="sc-bczRLJ iycqUS"
  data-testid="Testimonials"
  id="testimonials"
>
  <img
    alt="arrow left"
    class="sc-ftvSup iGaBgC"
    data-testid="arrow left"
    src="arrow-left-pink.svg"
  />
  <div
    class="sc-gsnTZi gvZlNX"
  >
    <h2>
      Testimonials
    </h2>
    <p
      class="sc-dkzDqf fuCwUC"
      data-testid="Quote"
    >
      this is a test quote
    </p>
    <div
      class="sc-hKMtZM Jbhfe"
      data-testid="Person"
    >
      Jane Doe
    </div>
    <div
      class="sc-eCYdqJ isYhdA"
      data-testid="Title"
    >
      Manager at Slalom Conuslting LLC
    </div>
    <div
      class="sc-jSMfEi iOloSv"
      data-testid="Dots"
    >
      <div
        class="sc-gKXOVf kCMRve"
      />
      <div
        class="sc-gKXOVf kJlTwh"
      />
    </div>
  </div>
  <img
    alt="arrow right"
    class="sc-papXJ cxbqPr"
    data-testid="arrow right"
    src="arrow-right-pink.svg"
  />
  <div
    class="sc-iBkjds dvtkjr"
  >
    <div
      class="sc-kDDrLX bdMEtj"
    >
      <div
        class="sc-jqUVSM bxNMxE"
      />
      <img
        alt="arrow left"
        class="sc-ftvSup hzntBy"
        data-testid="arrow left"
        src="arrow-left-pink.svg"
      />
    </div>
    <div
      class="sc-kDDrLX bdMEtj"
    >
      <div
        class="sc-jqUVSM bxNMxE"
      />
      <img
        alt="arrow right"
        class="sc-papXJ iMvUOL"
        data-testid="arrow right"
        src="arrow-right-pink.svg"
      />
    </div>
  </div>
</div>
`;
